artifactory/commands/python/poetry.go

```diff
@@ -0,0 +1,295 @@
+package python
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/jfrog/build-info-go/build"
+	"github.com/jfrog/build-info-go/entities"
+	buildinfoutils "github.com/jfrog/build-info-go/utils"
+	"github.com/jfrog/build-info-go/utils/pythonutils"
+	gofrogcmd "github.com/jfrog/gofrog/io"
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/commands/python/dependencies"
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/spf13/viper"
+)
+
+type PoetryCommand struct {
+	PythonCommand
+	poetryConfigRepoName string
+}
+
+const (
+	baseConfigRepoName     = "jfrog-server"
+	poetryConfigAuthPrefix = "http-basic."
+	poetryConfigRepoPrefix = "repositories."
+	pyproject              = "pyproject.toml"
+	pyprojectBackup        = "pyproject.toml.backup"
+)
+
+func NewPoetryCommand() *PoetryCommand {
+	return &PoetryCommand{
+		PythonCommand:        PythonCommand{pythonTool: pythonutils.Poetry},
+		poetryConfigRepoName: fmt.Sprintf("%s-%s", baseConfigRepoName, strconv.FormatInt(time.Now().UnixMilli(), 10)),
+	}
+}
+
+func (pc *PoetryCommand) Run() (err error) {
+	log.Info(fmt.Sprintf("Running Poetry %s.", pc.commandName))
+	var buildConfiguration *utils.BuildConfiguration
+	pc.args, buildConfiguration, err = utils.ExtractBuildDetailsFromArgs(pc.args)
+	if err != nil {
+		return err
+	}
+	pythonBuildInfo, err := utils.PrepareBuildPrerequisites(buildConfiguration)
+	if err != nil {
+		return
+	}
+	defer func() {
+		if pythonBuildInfo != nil && err != nil {
+			e := pythonBuildInfo.Clean()
+			if e != nil {
+				err = errors.New(err.Error() + "\n" + e.Error())
+			}
```

Comment:
Let's simplify to - 
```
	if err == nil {
		err = errorutils.CheckError(e)
	}
```

Reply:
We are cleaning up here only if err != nil. If the cleaning itself led to an error we aggregate the errors.
I can't see how it can be simplified.
Do you prefer we ignore the second error?